src/elements/content-explorer/ItemGridCell.js

```diff
@@ -0,0 +1,94 @@
+// @flow
+import * as React from 'react';
+import getSize from '../../utils/size';
+import moreOptionsCellRenderer from './moreOptionsCellRenderer';
+import dateCellRenderer from './dateCellRenderer';
+import nameCellRenderer from '../common/item/nameCellRenderer';
+import { getIcon } from '../common/item/iconCellRenderer';
+import './ItemGridCell.scss';
+
+type Props = {
+    canDelete: boolean,
+    canDownload: boolean,
+    canPreview: boolean,
+    canRename: boolean,
+    canShare: boolean,
+    isSmall: boolean,
+    isTouch: boolean,
+    item: ?BoxItem,
+    onItemClick: (item: BoxItem | string) => void,
+    onItemDelete: (item: BoxItem) => void,
+    onItemDownload: (item: BoxItem) => void,
+    onItemPreview: (item: BoxItem) => void,
+    onItemRename: (item: BoxItem) => void,
+    onItemSelect: (item: BoxItem, callback: Function) => void,
+    onItemShare: (item: BoxItem) => void,
+    rootFolderId: string,
+    view: View,
+};
+
+const ItemGridCell = ({
+    canDelete,
+    canDownload,
+    canPreview,
+    canRename,
+    canShare,
+    isSmall,
+    isTouch,
+    item,
+    onItemClick,
+    onItemDelete,
+    onItemDownload,
+    onItemPreview,
+    onItemRename,
+    onItemSelect,
+    onItemShare,
+    rootFolderId,
+    view,
+}: Props) => {
+    if (!item) {
+        return null;
+    }
+
+    const MoreOptionsCell = moreOptionsCellRenderer(
+        canPreview,
+        canShare,
+        canDownload,
+        canDelete,
+        canRename,
+        onItemSelect,
+        onItemDelete,
+        onItemDownload,
+        onItemRename,
+        onItemShare,
+        onItemPreview,
+        isSmall,
+    );
+
+    const NameCell = nameCellRenderer(
+        rootFolderId,
+        view,
+        onItemClick,
+        onItemSelect,
+        canPreview,
+        isSmall, // shows details if false
+        isTouch,
+    );
+
+    const DateCell = dateCellRenderer();
+
+    return (
+        <figure>
+            <div className="bce-ItemGridCell-itemThumbnail">
+                <div className="bce-ItemGridCell-itemIcon"> {getIcon(128, item)} </div>
+            </div>
+            <figcaption>
+                <NameCell rowData={item} />
+                <div> {getSize(item.size)} </div>
```

Comment:
Are the non-breaking spaces used intentionally here for visual spacing? If so, they should be replaced with CSS padding. If not, they should still be removed to avoid rendering issues; same with `getIcon` above.